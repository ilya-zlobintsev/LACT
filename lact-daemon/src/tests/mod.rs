use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
