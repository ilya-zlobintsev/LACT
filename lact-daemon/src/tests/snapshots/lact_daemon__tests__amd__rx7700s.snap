---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": null,
  "info": {
    "driver": "amdgpu",
    "link_info": {
      "current_speed": "16.0 GT/s PCIe",
      "current_width": "8"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Navi 33 [Radeon RX 7600/7600 XT/7600M XT/7600S/7700S / PRO W7600]",
        "model_id": "7480",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model_id": "0007",
        "vendor": "Framework Computer Inc.",
        "vendor_id": "F111"
      }
    },
    "vbios_version": "113-BRT125778.001"
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Navi 33 [Radeon RX 7600/7600 XT/7600M XT/7600S/7700S / PRO W7600]",
      "model_id": "7480",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model_id": "0007",
      "vendor": "Framework Computer Inc.",
      "vendor_id": "F111"
    }
  },
  "power_profile_modes": {
    "active": 0,
    "modes": {
      "0": {
        "components": [],
        "name": "BOOTUP_DEFAULT"
      }
    },
    "value_names": []
  },
  "stats": {
    "busy_percent": 0,
    "clockspeed": {
      "gpu_clockspeed": 0,
      "vram_clockspeed": 96
    },
    "core_power_state": 1,
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "pwm_current": 76,
      "speed_current": 0,
      "speed_max": 4900,
      "speed_min": 0
    },
    "memory_power_state": 0,
    "power": {
      "average": 1.0,
      "cap_current": 100.0,
      "cap_default": 100.0,
      "cap_max": 120.0,
      "cap_min": 100.0
    },
    "temps": {
      "edge": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 34.0
      },
      "junction": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 34.0
      },
      "mem": {
        "crit": 105.0,
        "crit_hyst": -273.1499938964844,
        "current": 42.0
      }
    },
    "voltage": {
      "gpu": 18
    },
    "vram": {}
  }
}
